test/Integration/SqlTriggerBindingIntegrationTests.cs

```diff
@@ -656,6 +656,72 @@ public async void LastAccessTimeColumn_Created_OnStartup()
             Assert.True(1 == (int)this.ExecuteScalar("SELECT 1 FROM sys.columns WHERE Name = N'LastAccessTime' AND Object_ID = Object_ID(N'[az_func].[GlobalState]')"), $"{GlobalStateTableName} should have {LastAccessTimeColumnName} column after restarting the listener.");
         }
 
+        /// <summary>
+        /// Tests that trigger function executes on table whose name is a reserved word (User).
+        /// </summary>
+        [Theory]
+        [SqlInlineData()]
+        [UnsupportedLanguages(SupportedLanguages.Java)] // test timing out for Java
+
+        public async void ReservedTableNameTest(SupportedLanguages lang)
+        {
+            this.SetChangeTrackingForTable("User");
+            this.StartFunctionHost(nameof(ReservedTableNameTrigger), lang, true);
+            User expectedResponse = Utils.JsonDeserializeObject<User>(/*lang=json,strict*/ "{\"UserId\":999,\"UserName\":\"test\",\"FullName\":\"Testy Test\"}");
+            int index = 0;
+            string messagePrefix = "SQL Changes: ";
+
+            var taskCompletion = new TaskCompletionSource<bool>();
+
+            void MonitorOutputData(object sender, DataReceivedEventArgs e)
+            {
+                if (e.Data != null && (index = e.Data.IndexOf(messagePrefix, StringComparison.Ordinal)) >= 0)
+                {
+                    string json = e.Data[(index + messagePrefix.Length)..];
```

Comment:
We have this logic in a couple of places now, could you make a issue to follow up on for consolidating this into a single helper?